lucet-runtime/lucet-runtime-macros/src/lib.rs

```diff
@@ -39,6 +39,12 @@ pub fn lucet_hostcall(_attr: TokenStream, item: TokenStream) -> TokenStream {
     };
 
     let mut hostcall = syn::parse_macro_input!(item as syn::ItemFn);
+    if let Some(abi) = &hostcall.sig.abi {
+        let err = quote_spanned! {abi.span()=>
+            compile_error!("Functions annotated with `#[lucet_hostcall]` may not have an `extern` ABI");
```

Comment:
dear future readers: if you're `extern "Rust"`ing and this incorrectly rejects your program, please say hello! also, why are you using `extern "Rust"`!